library/bignum_mod.h

```diff
@@ -199,7 +199,35 @@ int mbedtls_mpi_mod_sub( mbedtls_mpi_mod_residue *X,
 /* END MERGE SLOT 4 */
 
 /* BEGIN MERGE SLOT 5 */
-
+/**
+ * \brief Perform a fixed-size modular addition.
+ *
+ * Calculate `A + B modulo N`.
+ *
+ * \p A, \p B and \p X must all have the same number of limbs as \p N.
+ *
+ * \p X may be aliased to \p A or \p B, or even both, but may not overlap
+ * either otherwise.
+ *
+ * \note This function does not check that \p A or \p B are in canonical
+ *       form (that is, are < \p N) - that will have been done by
+ *       mbedtls_mpi_mod_residue_setup().
+ *
+ * \param[out] X    The address of the result MPI. Must be initialized.
+ *                  Must have the same number of limbs as the modulus \p N.
```

Comment:
It shouldn't matter. Adding numbers is the same whether they are in Montgomery form or not

Reply:
At this level of the API we should be more restrictive. In the residue lifecycle when they are created they are associated with a specific modulus. Only operations relating to this modulus are guaranteed to work. In practice it might work with other moduli as well, but the `_mod` API only needs to promise to yield correct result if the operands are associated the modulus of the operation.